acareeg/eegip.py

```diff
@@ -171,53 +175,122 @@ def process_events_london_resting_state(raw, age, tmax=1):
 
         if len(rst0) == 0 and len(rst1) == 0:
             return None
+        assert(len(rst0) == len(rst1))
 
-        annot_sample = np.concatenate([np.arange(start, stop - 0.999, 1.0) for start, stop in zip(rst0, rst1)])
-        annot_sample = (annot_sample * freq).astype(int)
+        # Keep only trials that are at least 10 s long.
+        # All trials are defined by Rst0-Rst1 pair of events.
+        annot_sample = np.concatenate([np.arange(start*freq, (stop-tmax)*freq, tmax*freq)
+                                       for start, stop in zip(rst0, rst1)
+                                       if stop - start > 10.0]).astype(int)
         annot_id = [1] * len(annot_sample)
 
     elif age == 12:
-        annots = [a for a in raw.annotations if a["description"] in ["eeg1", "eeg2", "eeg3"]]
-        if len(annots) == 0:
+        """
+         For London 12m, resting-state data are marked only by Rst0 markers at the begining of
+         every trials. Trials and consecutives, so we can use the following Rst0 maker as being the 
+         end of the current trials. The last trials is ended with the last occurence of a SWIR or SWEN
+         event. 
+         
+         Trials shorter than 10s are rejected. Trials longer than 50s are truncated at 50s.
+         
+         Events eeg1, eeg2 and eeg3, when present, are at the same time as the Rst0 events. They 
+         are used to set the type of event (type of video that was presented). If no eegx event
+         is available for a Rst0, as type eeg4 is set, which indicate a resting-state event using
+         an unknown video.
+        """
+
+        onsets = [annot["onset"] for annot in raw.annotations if annot["description"] == "Rst0"]
+        if len(onsets) == 0:
             return None
 
-        if raw.annotations[-1]["onset"] > annots[-1]["onset"]:
```

Comment:
I dont understand the condition of this if statement.

Reply:
ok actually...; "if the last Rst0 onset occurs after the onset of the last event (i.e. swin/swir), THEN below. (?)

Reply:
The "-" sign and the red highlighting tell you that this is a portion of the code that has been removed by this commit... so it is not that important for you to try to understand code that has been removed at this point I think.